src/main/java/uk/gov/hmcts/reform/hmc/service/LinkedHearingGroupServiceImpl.java

```diff
@@ -96,6 +101,57 @@ public void updateLinkHearing(String requestId, HearingLinkGroupRequest hearingL
         validateHearingLinkGroupRequestForUpdate(requestId, hearingLinkGroupRequest);
     }
 
+    @Override
+    public LinkedHearingGroupResponses getLinkedHearingGroupDetails(String requestId) {
+        LinkedGroupDetails linkedGroupDetails =
+            linkedGroupDetailsRepository.getLinkedGroupDetailsByRequestId(requestId);
+        if (linkedGroupDetails != null) {
+            return getLinkedHearingGroupDetails(linkedGroupDetails);
+        } else {
+            throw new LinkedGroupNotFoundException(requestId,INVALID_LINKED_GROUP_REQUEST_ID_DETAILS);
+        }
+    }
+
+    private LinkedHearingGroupResponses getLinkedHearingGroupDetails(LinkedGroupDetails linkedGroupDetails) {
+        LinkedHearingGroupResponses responses = new LinkedHearingGroupResponses();
+        setGroupDetails(linkedGroupDetails,responses);
+        setHearingsInGroup(linkedGroupDetails.getLinkedGroupId(),responses);
```

Comment:
It's probably better on these two methods to not pass the `responses` object in, and instead do something like:
```suggestion
        responses.setGroupDetails(getGroupDetails(linkedGroupDetails));
        responses.setHearingsInGroup(getHearingsInGroup(linkedGroupDetails.getLinkedGroupId()));
```